include/cppkafka/header_list.h

```diff
@@ -159,10 +165,14 @@ class HeaderList {
      */
     explicit operator bool() const;
     
+    /**
+     * \brief Indicates if this list owns the underlying handle or not.
+     */
+    bool is_owning() const;
```

Comment:
I don't think this should be exposed. Owning handles can only be created privately so nobody from the outside knows that handles can be non owning. Also see my comment on the place where this is used, as I don't think we even need this (unless there's a use for it outside the library itself).

Reply:
See my comment above. I can remove it if you like.

Reply:
I take back the "can only be created privately", that's not true. Anyway, let's follow this on the other comment.